src/librustc_mir/borrow_check/nll/subtype_constraint_generation.rs

```diff
@@ -112,23 +108,19 @@ impl<'cx, 'tcx> SubtypeConstraintGenerator<'cx, 'tcx> {
                 // reverse direction, because `regioncx` talks about
                 // "outlives" (`>=`) whereas the region constraints
                 // talk about `<=`.
-                self.regioncx.add_outlives(span, b_vid, a_vid, at_location);
+                self.regioncx.add_outlives(span, b_vid, a_vid);
```

Comment:
Am I right in my understanding that one side-effect of no longer passing `at_location` into `add_outlives` is that our diagnostics stop being able to point to this location as the place where a particular constraint was introduced, which is what is leading to problems like the regression in diagnostic quality for e.g. https://github.com/rust-lang/rust/pull/50938/commits/a904c55357ae48d85bc8f0159d6a456dba6b5792#diff-40124ba66f06953794b406e3bb131b95L6 ?)

(And if that hypothesis is correct, is your thinking that we're going to have to solve that problem more generally anyway, perhaps as part of #51188, but that finding a solution to *that* problem should not block landing PR's like this one?)

Reply:
hey @nikomatsakis do you have an answer to my Q above ?